### DISCUSSION TITLE
Search for filenames

### DISCUSSION THREAD

**Discussion by mleonhard-flexport** [community/community#35148] [Other Feature Feedback, Questions, & Ideas]
Please let us search for filenames across an organization or repository.  I do this frequently and am disappointed each time when the results list does not contain existing files with matching names.

Using only the `path` search operator yields the error
> Search text is required when searching source code. Searches that use qualifiers only are not allowed.

Presumably, we could add a keyword that is present in the file ("function") and then the `path` operator would work.  But none of these work:
* `function path:something.js`
* `function path:/.*/something\.js/`
* `function path:something\.js$` (and other examples from https://cs.github.com/about/syntax#path )

**Comment by look** [comment]
Hi, I'm not sure I understand the problem. In the new Code Search, you can search for results by path alone, and you can limit the results to an organization or repository. Here are some examples:

https://cs.github.com/?scopeName=All+repos&scope=&q=org%3Aapache+path%3AREADME.md

https://cs.github.com/?scopeName=All+repos&scope=&q=repo%3Aapache%2Fkafka+path%3AREADME.md

Are you using the current code search or the Technology Preview?

**Reply by 3kh0** [reply]
Thank you @look this helped me with my issue!

**Reply by mleonhard-flexport** [reply]
I guess the problem is limited to GitHub Enterprise.  We're using GitHub Enterprise Server 3.5.1.  The repository has a file at https://github.flexport.io/flexport/flexport/blob/master/README.md but searching for `path:README.md` "In this repository" https://github.flexport.io/flexport/flexport/search?q=path%3AREADME.md yields

> We couldn’t find any code matching 'path:Readme.md' in [flexport/flexport](https://github.flexport.io/flexport/flexport)

Searching for `path:README.md` in "All GitHub Enterprise" https://github.flexport.io/search?q=path%3AREADME.md&type=code yields

> We could not perform this search
> Search text is required when searching source code. Searches that use qualifiers only are not allowed. Were you searching for something else?

And searching for `the path:README.md` "All GitHub Enteprise" https://github.flexport.io/search?q=the+path%3AREADME.md yields

> We couldn’t find any code matching 'the path:README.md'
